Static import of @tanstack/react-query-devtools breaks consumer builds

### Description

`src/Main.tsx` uses a static top-level import for `@tanstack/react-query-devtools`:

```ts
import { ReactQueryDevtools } from '@tanstack/react-query-devtools';
```

Even though the component is only rendered conditionally in development:

```tsx
{ getAppConfig(appId).NODE_ENV === 'development' && <ReactQueryDevtools initialIsOpen={false} /> }
```

Webpack resolves all static imports at build time, regardless of runtime conditions. This means consumers of this package must have `@tanstack/react-query-devtools` installed — even for production builds — or their build fails with:

```
Module not found: Error: Can't resolve '@tanstack/react-query-devtools'
```

Since `@tanstack/react-query-devtools` is not listed in this package's `dependencies`, any consumer that doesn't happen to have it installed elsewhere will hit this error.

### Suggested fix

Replace the static import with a dynamic one so webpack can code-split it and the package only needs to be available in development environments:

```tsx
// Instead of a static import, load it lazily
const ReactQueryDevtools = React.lazy(() =>
  import('@tanstack/react-query-devtools').then(m => ({ default: m.ReactQueryDevtools }))
);
```

With a dynamic import, `@tanstack/react-query-devtools` can legitimately live in `devDependencies` and consumers won't need it at all.

--- a/src/Main.tsx
+++ b/src/Main.tsx
@@ -1,12 +1,19 @@
+import { lazy, Suspense } from 'react';
 import { CurrentAppProvider, getAppConfig } from '@openedx/frontend-base';
 import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
 import { Outlet } from 'react-router-dom';
-import { ReactQueryDevtools } from '@tanstack/react-query-devtools';
 import { AlertProvider } from './providers/AlertProvider';
 import { appId } from './constants';
 import PageWrapper from './pageWrapper/PageWrapper';
 
 import './app.scss';
+
+// Use a dynamic import guarded by process.env.NODE_ENV so the consumer's
+// webpack dead-code-eliminates this in production builds, meaning
+// @tanstack/react-query-devtools does not need to be installed by consumers.
+const ReactQueryDevtools = process.env.NODE_ENV === 'development'
+  ? lazy(() => import('@tanstack/react-query-devtools').then((m) => ({ default: m.ReactQueryDevtools })))
+  : null;
 
 const queryClient = new QueryClient();
 
@@ -18,7 +25,11 @@ const Main = () => (
           <PageWrapper>
             <Outlet />
           </PageWrapper>
-          { getAppConfig(appId).NODE_ENV === 'development' && <ReactQueryDevtools initialIsOpen={false} /> }
+          {ReactQueryDevtools && getAppConfig(appId).NODE_ENV === 'development' && (
+            <Suspense fallback={null}>
+              <ReactQueryDevtools initialIsOpen={false} />
+            </Suspense>
+          )}
         </main>
       </AlertProvider>
     </QueryClientProvider>

